File: src/main/java/net/matees/settings/global/WorldBorderSize.java
package net.matees.settings.global;

import org.bukkit.Bukkit;
import org.bukkit.Material;

import net.matees.settings.Global;
import net.matees.settings.IntegerSetting;

public class WorldBorderSize extends IntegerSetting implements Global {
    private static final WorldBorderSize INSTANCE = new WorldBorderSize();
    private int setting = 500;

    private WorldBorderSize() {
    }

    public static WorldBorderSize getInstance() {
        return INSTANCE;
    }

    @Override
    public String getName() {
        return "World Border Size";
    }

    @Override
    public String getDescription() {
        return "Set world border size";
    }

    @Override
    public Integer getSetting() {
        return this.setting;
    }

    @Override
    public void setSetting(Integer setting) {
        this.setting = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.IRON_BARS;
    }

    @Override
    public int getMenuItemSlot() {
        return 11;
    }

    @Override
    public void onChange() {
        Bukkit.getWorld("world").getWorldBorder().setSize((double) this.getSetting());
    }

}

File: src/main/java/net/matees/settings/global/WorldBorderSize.kt
package net.matees.settings.global

import net.matees.settings.AbstractIntegerSetting
import net.matees.settings.IGlobal
import org.bukkit.Bukkit
import org.bukkit.Material

class WorldBorderSize private constructor() : AbstractIntegerSetting(), IGlobal {
    override var setting: Int? = 500

    override val min: Int
        get() = 50

    override val name: String
        get() = "World Border Size"

    override val description: String
        get() = "Set world border size"

    override val menuItemMaterial: Material?
        get() = Material.IRON_BARS

    override val menuItemSlot: Int
        get() = 11

    override fun onChange() {
        Bukkit.getWorld("world")!!.worldBorder.size = setting!!.toDouble()
    }

    companion object {
        val instance: WorldBorderSize = WorldBorderSize()
    }
}
